.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,31 @@
+name: Docker build workflow
+# NB: This will only run on PRs from the main repo and not forks
+# https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull-request-events-for-forked-repositories
+on:
+  pull_request:
+    branches:
+      - master
+jobs:
+  build-docker:
+    name: Build and push Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      # NB: For PRs, this will create two Docker tags: sha-${GIT_SHA} and pr-${PR_NUMBER}
+      # https://github.com/crazy-max/ghaction-docker-meta#overview
+      - id: docker_meta
+        uses: crazy-max/ghaction-docker-meta@v1
+        with:
+          images: gcr.io/xpn-heroic-1/heroic
```

Comment:
Given this is a public repo, should images for PRs push to internal GCR, or Docker Hub?

If i'm understanding Docker Meta properly, this prepares for a push to internal GCR.

Reply:
Right, it pushes to the GCR repo which is internal only. I set it up for GCR because I think the PR builds aren't useful for general consumption, and I wanted to avoid the new Docker Hub pull limits. But its a weak opinion, happy to switch it if you disagree.

Reply:
I agree with the decision since the use case is for internal testing, and we'd have to take an additional step to then push to GCR which makes the process unnecessarily complicated. Not to mention, PR builds could be broken/faulty, so best to not publicize them.